diff --git a/conda/shell/etc/profile.d/conda.sh b/conda/shell/etc/profile.d/conda.sh
index 544cce4dc..c9e10949b 100644
--- a/conda/shell/etc/profile.d/conda.sh
+++ b/conda/shell/etc/profile.d/conda.sh
@@ -29,7 +29,7 @@ __conda_activate() {
 
 __conda_reactivate() {
     # FUTURE: conda 25.9, remove this function
-    echo `__conda_reactivate` is deprecated and will be removed in 25.9. Use `__conda_activate reactivate` instead. 1>&2
+    echo "'__conda_reactivate' is deprecated and will be removed in 25.9. Use '__conda_activate reactivate' instead." 1>&2
     __conda_activate reactivate
 }
 
