From 62196c897df3d7aea7063d0c08d1bf6e6fd91600 Mon Sep 17 00:00:00 2001
From: "Benjamin A. Beasley" <code@musicinmybrain.net>
Date: Fri, 2 Aug 2024 08:50:43 -0400
Subject: [PATCH 1/2] Adapt for logging lock internal changes in Python 3.13

---
 conda/common/io.py | 9 +++++++--
 1 file changed, 7 insertions(+), 2 deletions(-)

diff --git a/conda/common/io.py b/conda/common/io.py
index 91f37e144d0..d3b655e47f4 100644
--- a/conda/common/io.py
+++ b/conda/common/io.py
@@ -269,11 +269,16 @@ def argv(args_list):
 
 @contextmanager
 def _logger_lock():
-    logging._acquireLock()
+    try:
+        # Python 3.13+
+        acquire, release = logging._prepareFork, logging._afterFork
+    except AttributeError:
+        acquire, release = logging._acquireLock, logging._releaseLock
+    acquire()
     try:
         yield
     finally:
-        logging._releaseLock()
+        release()
 
 
 @contextmanager

From 78cd209128292371d9d2abbca7b3d1f4912d1c49 Mon Sep 17 00:00:00 2001
From: "Benjamin A. Beasley" <code@musicinmybrain.net>
Date: Mon, 5 Aug 2024 10:52:49 -0400
Subject: [PATCH 2/2] Add news item for PR#14117

---
 news/14117-logging-lock-changes | 19 +++++++++++++++++++
 1 file changed, 19 insertions(+)
 create mode 100644 news/14117-logging-lock-changes

diff --git a/news/14117-logging-lock-changes b/news/14117-logging-lock-changes
new file mode 100644
index 00000000000..7f708aa0eed
--- /dev/null
+++ b/news/14117-logging-lock-changes
@@ -0,0 +1,19 @@
+### Enhancements
+
+* <news item>
+
+### Bug fixes
+
+* <news item>
+
+### Deprecations
+
+* <news item>
+
+### Docs
+
+* <news item>
+
+### Other
+
+* Adapt for logging lock internal changes in Python 3.13 (#14117)
